src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(health = 100, name) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(name, health);
+  }
+
+  bite(victim) {
+    if (victim instanceof Herbivore) {
+      if (!victim.hidden) {
+        victim.health -= 50;
+      }
+    }
+
+    if (victim.health <= 0) {
```

Comment:
HI) You can use === rather than <= because 'healthy' will never be lower than 0.

Reply:
Not critical, but relevant comment)